### PR TITLE
release raiwidgets and responsibleai 0.15.0

### PR DESCRIPTION
**Pull request by imatiach-msft**
## v0.15.0

- new features
  - ## Model Analysis
    - move raw data into /data directory
    - perform schema validations during compute and load
    - rename model analysis to rai insights
    - rename ModelAnalysisDashboard to ResponsibleAIDashboard
    - directory based state management for model analysis tools
    - add .json extension to train and test dataset
  - ## Error Analysis
    - port ErrorReport schema validation for error analysis to responsibleai
    - add error analysis error importances (aka correlation of features with error) to model analyzer API
    - add support for special classification outcome filter to error analysis backend to work with model analysis filters
  - ## Counterfactual
    - remove save as new data point option for read-only mode on What-if counterfactual
    - split counterfactual output into individual json files
    - add support for non-string categorical values
- bug fixes
  - ## Model Analysis
    - fix cohort text out of button box
    - fix bug when the tree node has 0 errors and N correct, the raw data table shows the opposite with N+1 errors and -1 correct.
    - when classes are not passed, sort them automatically in RAIInsights
    - fix composite filter issues
  - ## Counterfactual
    - fix sort feature columns by counterfactual
  - ## Error Analysis
    - fix map shift text not being clear, provide conditional dialog text based on current view
    - fix tree map connection line doubled after switching to dark theme
- ## other
  - update readme links to be absolute instead of relative path
  - add more UX IFI tests for responsibleaitoolbox-classification-model-assessment notebook
  - several E2E UX test fixes
  - add E2E UX tests for Feature Importance - Cohort functionality
  - add test to render ModelAnalysisDashboard after save and load
  - add test to render ResponsibleAIDashboard after save and load
  - add E2E tests for What-if Counterfactuals - common functionalities & Y Axis Flyout
  - add separate github workflow for python linting

### CODE REVIEW

File: responsibleai/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 dice-ml>=0.7.2,<0.8
 econml~=0.12.0
 jsonschema
-erroranalysis>=0.1.27
+erroranalysis>=0.1.28
```
Comment (xuke444) [inline responsibleai/requirements.txt:4]: should we also have an upper bound for all packages?
At least to next major version?

Comment (imatiach-msft) [reply responsibleai/requirements.txt:4]: erroranalysis has always been backwards compatible, so no upper bound has been needed so far

File: CHANGES.md [hunk 2]
```diff
@@ -24,6 +24,40 @@ this file to understand what changed.
...
+    - rename model analysis to rai insights
+    - rename ModelAnalysisDashboard to ResponsibleAIDashboard
+    - directory based state management for model analysis tools
+    - add .json extension to train and test dataset
+  - ## Error Analysis
+    - port ErrorReport schema validation for error analysis to responsibleai
+    - add error analysis error importances (aka correlation of features with error) to model analyzer API
+    - add support for special classification outcome filter to error analysis backend to work with model analysis filters
+  - ## Counterfactual
+    - remove save as new data point option for read-only mode on What-if counterfactual
```
Comment (gaugup) [inline CHANGES.md:42]: I would add this change as well :- "Split counterfactual output into individual json files"

Comment (imatiach-msft) [reply CHANGES.md:42]: yes I will have to keep updating it for now until all necessary PRs have been merged

Comment (imatiach-msft) [reply CHANGES.md:42]: done!

**Review by gaugup — APPROVED** [review]

**Review by xuke444 — APPROVED** [review]

**Pull request merged**